### PR TITLE
coreaction: init at 4.2.0

### PR DESCRIPTION
**Pull request by dan4ik605743**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dan4ik605743** [comment]
I do not understand what the problem is ..

**Comment by dan4ik605743** [comment]
image
😕

**Comment by OPNA2608** [comment]
Your `libcsys` package is identical in code to your `coreaction` one, which has `libcsys` in its `buildInputs`. So `libcsys` tried to depend on itself -> infinite recursion.

**Comment by dan4ik605743** [comment]
> Your `libcsys` package is identical in code to your `coreaction` one, which has `libcsys` in its `buildInputs`. So `libcsys` tried to depend on itself -> infinite recursion.

Oh yo. I did not see at all. Not that package loaded. Thank you so much.

**Comment by OPNA2608** [comment]
Result of `nixpkgs-review pr 131024` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>coreaction</li>
    <li>libcsys</li>
  </ul>
</details>

---

`coreapps` prints this error:
```
Settings file  "/home/bt1cn/.config/coreapps/coreapps.conf"
whats this  "/home/bt1cn/.config/cubocore/"
Plugin Error: "The shared library was not found." "/nix/store/sz9ryk99kfh7d5qrxb721nh0rybymkq0-coreaction-4.2.0/lib/coreapps/plugins/libbatery.so"
```

[`fetchpatch`ing this commit](https://gitlab.com/cubocore/coreapps/coreaction/-/commit/1d1307363614a117978723eaad2332e6e8c05b28.patch) or doing the following + deleting `~/.config/coreapps/` fixes the error & subsequently the battery widget:

```nix
postPatch = ''
  substituteInPlace app/settings.cpp \
    --replace 'libbatery.so' 'libbattery.so'
'';
```

<details>
  <summary>With fetchpatch (also adds another widget)</summary>

grafik

</details>

**Comment by dan4ik605743** [comment]
> Result of `nixpkgs-review pr 131024` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 2 packages built:
> `coreapps` prints this error:
> 
> ```
> Settings file  "/home/bt1cn/.config/coreapps/coreapps.conf"
> whats this  "/home/bt1cn/.config/cubocore/"
> Plugin Error: "The shared library was not found." "/nix/store/sz9ryk99kfh7d5qrxb721nh0rybymkq0-coreaction-4.2.0/lib/coreapps/plugins/libbatery.so"
> ```
> 
> [`fetchpatch`ing this commit](https://gitlab.com/cubocore/coreapps/coreaction/-/commit/1d1307363614a117978723eaad2332e6e8c05b28.patch) or doing the following + deleting `~/.config/coreapps/` fixes the error & subsequently the battery widget:
> 
> ```nix
> postPatch = ''
>   substituteInPlace app/settings.cpp \
>     --replace 'libbatery.so' 'libbattery.so'
> '';
> ```
> 
> With fetchpatch (also adds another widget)

Well now add. Thank you.

**Comment by dan4ik605743** [comment]
> Result of `nixpkgs-review pr 131024` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
> 
> 2 packages built:
> `coreapps` prints this error:
> 
> ```
> Settings file  "/home/bt1cn/.config/coreapps/coreapps.conf"
> whats this  "/home/bt1cn/.config/cubocore/"
> Plugin Error: "The shared library was not found." "/nix/store/sz9ryk99kfh7d5qrxb721nh0rybymkq0-coreaction-4.2.0/lib/coreapps/plugins/libbatery.so"
> ```
> 
> [`fetchpatch`ing this commit](https://gitlab.com/cubocore/coreapps/coreaction/-/commit/1d1307363614a117978723eaad2332e6e8c05b28.patch) or doing the following + deleting `~/.config/coreapps/` fixes the error & subsequently the battery widget:
> 
> ```nix
> postPatch = ''
>   substituteInPlace app/settings.cpp \
>     --replace 'libbatery.so' 'libbattery.so'
> '';
> ```
> 
> With fetchpatch (also adds another widget)

@OPNA2608 I understood correctly. Add fetchpath and postpath?

**Comment by OPNA2608** [comment]
You can do either
```nix
patches = [(fetchpatch {
  url = "https://gitlab.com/cubocore/coreapps/coreaction/-/commit/1d1307363614a117978723eaad2332e6e8c05b28.patch";
  sha256 = "039x19rsm23l9vxd5mnbl6gvc3is0igahf47kv54v6apz2q72l3f";
})];
```
which fixes the problem + adds another widget, **or** the `postPatch` way, which fixes only the typo without introducing any new features into the 4.2.0 build.

**Comment by dan4ik605743** [comment]
> > I understood correctly. Add fetchpath and postpatch?
> 
> You can do either
> 
> ```nix
> patches = [(fetchpatch {
>   url = "https://gitlab.com/cubocore/coreapps/coreaction/-/commit/1d1307363614a117978723eaad2332e6e8c05b28.patch";
>   sha256 = "039x19rsm23l9vxd5mnbl6gvc3is0igahf47kv54v6apz2q72l3f";
> })];
> ```
> 
> which fixes the problem + adds another widget, **or** the `postPatch` way, which fixes only the typo without introducing any new features into the 4.2.0 build.

Yes, I did it.

File: pkgs/applications/misc/coreaction/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-5qEZNLvbgLoAOXij0wXoVw2iyvytsYZikSJDm6F6ddc=";
+  };
+
+  patches = [
+    ## Fix Plugin Error: "The shared library was not found." "libbatery.so"
+    (fetchpatch {
+      url = "https://gitlab.com/cubocore/coreapps/coreaction/-/commit/1d1307363614a117978723eaad2332e6e8c05b28.patch";
+      sha256 = "0rs8j54d5zf0m6iwsdfmxqakv63dyb4mfbwl9ghk85ffq95fnsn9";
```
Comment (OPNA2608) [inline pkgs/applications/misc/coreaction/default.nix:18]: I'm getting a different hash here.
```
hash mismatch in fixed-output derivation '/nix/store/swiyi93ffa12k5hw1xraby6pq7hvs2ma-1d1307363614a117978723eaad2332e6e8c05b28.patch':
  wanted: sha256:0rs8j54d5zf0m6iwsdfmxqakv63dyb4mfbwl9ghk85ffq95fnsn9
  got:    sha256:039x19rsm23l9vxd5mnbl6gvc3is0igahf47kv54v6apz2q72l3f
```

Comment (dan4ik605743) [reply pkgs/applications/misc/coreaction/default.nix:18]: ok

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Lastly, could you fix the commit order & contents?
- the dependency `libcsys` should be committed before the application `coreaction` that uses it, and
- the `libcsys` commit shouldn't contain the patch addition to `coreaction`, that belongs in the `coreaction` commit.

LGTM otherwise, x86_64 builds here and aarch64 builds on OfBorg.

**Comment by dan4ik605743** [comment]
> Lastly, could you fix the commit order & contents?
> 
> * the dependency `libcsys` should be committed before the application `coreaction` that uses it, and
> * the `libcsys` commit shouldn't contain the patch addition to `coreaction`, that belongs in the `coreaction` commit.
> 
> LGTM otherwise, x86_64 builds here and aarch64 builds on OfBorg.

Do not quite understand. Could you show on the example of my code.

**Comment by dan4ik605743** [comment]
@SuperSandro2000 Please check my PR

**Pull request merged**